libraries/oneview_helper.rb

```diff
@@ -18,31 +18,39 @@ module Helper
     # @param context Context from the resource action block (self)
     # @param [String, Symbol] type Name of the resource type. e.g., :EthernetNetwork
     # @param [String, Symbol] action Action method to call on the resource. e.g., :create_or_update
-    def self.do_resource_action(context, type, action)
-      klass = OneviewCookbook::Helper.get_resource_class(context, type)
+    # @param [Module] base_module The base API module supported. e.g. OneviewCookbook::ImageStreamer
+    def self.do_resource_action(context, type, action, base_module = OneviewCookbook)
+      klass = OneviewCookbook::Helper.get_resource_class(context, type, base_module)
       res = klass.new(context)
       res.send(action)
     end
 
     # Get the resource class by name, taking into consideration the resource's properties
     # @param context Context from the resource action block (self)
     # @param [String, Symbol] type Name of the resource type. e.g., :EthernetNetwork
+    # @param [Module] base_module The base API module supported. e.g. OneviewCookbook::ImageStreamer
     # @return [Class] Resource class
-    def self.get_resource_class(context, resource_type)
+    def self.get_resource_class(context, resource_type, base_module = OneviewCookbook)
       load_sdk(context)
-      api_version = context.property_is_set?(:api_version) ? context.api_version : context.node['oneview']['api_version']
-      api_module = get_api_module(api_version)
+      # Loads the api version if the property was specified directly
+      context_api_version = context.api_version if context.property_is_set?(:api_version)
+      # Loads the api version if it was specified in the client
+      client_api_version = convert_keys(context.client, :to_sym)[:api_version] if context.property_is_set?(:client)
+      # Loads the api version giving preference: property > client > node default
+      api_version = context_api_version || client_api_version || context.node['oneview']['api_version']
```

Comment:
> Especially if the user doesn't declare an API version, so it gets set to OneviewSDK::DEFAULT_API_VERSION (200).

So, my 10 cents on this as a new user of the module:
I actually expected it to be using the API I had declared for the client. 
At first, I was trying to just declare the client and call my resources.. and I was hitting errors because the resource did not exist for version 200, and that was something I felt was really counter-intuitive. I then had to actually declare in each resource the version I wanted it to use, instead of it 'in a smart way' getting the version from the client I was passing in.
Bottom line: as a new user, I'd expect it the api version to match my client, and if I actually want to differ from that client version, I'd then explicitly tell the resource which api version to use, not the other way around.